### PR TITLE
Service support implementation

### PR DESCRIPTION
**Pull request by shin-**
Services and tasks.

See: #1085

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Seems need to apply some fixes:
e.g.
https://github.com/docker/engine-api/commit/14c70cc34bfee8f1335137b77cc8d496d0169bf8
https://github.com/docker/engine-api/commit/2d2ff7c7d6ee8bc7408e7c2440a4da1cf309c198

**Comment by pacoxu** [comment]
Any update on this? 

we get error during update service
{"message":"Invalid service version '': strconv.ParseUint: parsing \"\": invalid syntax"}
as https://forums.docker.com/t/docker-swarm-scale-service-using-update-api/19589

**Comment by atakada** [comment]
Is there anyone still working on it? I found some parameter names to be updated, but I'll be happy if there are any branches I can try.
https://github.com/atakada/docker-py/tree/service-support-fix

**Comment by pacoxu** [comment]
@atakada 
Finally, we found that you have to use the update API like
POST /services/<service_id>/update?version=<current-version-index>
{
 "Name": "**"
"TaskTemplate" : {***}
}

Hope it can help you.

**Comment by shin-** [comment]
Thank you for your help, folks! I've been focused on other tasks lately but I should be able to pick this back up soon.

**Comment by atakada** [comment]
Thanks for comments&updates! Unfortunately I'm working for other stuff, but will come back.

**Comment by shin-** [comment]
- I have rebased this on top of #1147 so I could use the swarm init methods in the integration tests.
- Updated methods and added the missing `update_service`
- Moved the `TaskTemplate`, `ContainerSpec` and similar classes to the `docker.types` package.
- Started writing docs. API docs still need to be filled out.

At that point everything should work properly, but please leave a comment if you're testing this and running into issues. :)

File: tests/integration/service_test.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import random
+
+import docker
+# import pytest
```
Comment (dnephin) [inline tests/integration/service_test.py]: commented out import

File: docker/types/services.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                self['VolumeOptions'] = volume_opts
+            if propagation:
+                raise errors.DockerError(
+                    'Mount type is volume but `propagation` argument has been '
+                    'provided.'
+                )
+
+    @classmethod
+    def parse_mount_string(cls, string):
+        parts = string.split(':')
```
Comment (dnephin) [inline docker/types/services.py]: The CLI doesn't support this syntax for mounts, so I'm not sure it should be supported here

**Comment by dnephin** [comment]
LGTM

File: setup.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 SOURCE_DIR = os.path.join(ROOT_DIR)
 
 requirements = [
-    'requests >= 2.5.2',
+    'requests >= 2.5.2, < 2.11',
```
Comment (shin-) [inline setup.py]: https://github.com/docker/docker-py/pull/1204

Comment (glensc) [inline setup.py]: hmm, #1204 just removes the pin. so it's actually ok to use requests-2.11 without any other code changes?

Comment (shin-) [inline setup.py]: You can use any version >= 2.5.2, except 2.11.0. 
2.11.1 fixed the bug so that version or above is fine as well.

Comment (glensc) [inline setup.py]: thanks for clarification!

**Pull request merged**